### ISSUE TITLE
nav.pop fails to trigger if called too fast after alert is dismissed

### ISSUE THREAD

**Issue by wli** [driftyco/ionic#6185]
#### Short description of the problem:
`nav.pop()` fails to go to previous screen when called too fast after an alert is dismissed

#### What behavior are you expecting?
The app should go back to the previous screen. Instead, it stays on the same screen, and it gets buggier from there.

**Steps to reproduce:**
1. Create an alert that calls nav.pop() in a handler 100ms after the handler is triggered.

```
	pressSlow() { // okay
	  this.press(500);
	}
	
	pressFast() { // buggy
	  this.press(100);
	}
	
	press(timeout:number) {
	  let alert = Alert.create({
	    title: 'Hello',
	    message: 'This is an alert',
	    buttons: [{
	      text: 'Ok',
	      handler: () => {
	        let loading = Loading.create({
	          content: 'Loading...'
	        });
	        this.nav.present(loading);
	        
	        setTimeout(() => {
	          loading.dismiss();
	          this.nav.pop();
	        }, timeout);
	      }
	   }]
	  });
	  this.nav.present(alert);
	}
```

**Other information:** (e.g. stacktraces, related issues, suggestions how to fix, stackoverflow links, forum links, etc)
There are no console errors that get triggered. Sometimes, this will return in a black screen instead (in an app).

**Which Ionic Version?**
2.0.0-beta.4

#### Plunker that shows an example of your issue
http://plnkr.co/edit/Ses1tp80H8UTK4FBqheE?p=preview

**Run `ionic info` from terminal/cmd prompt:** (paste output below)
```
Cordova CLI: 6.1.1
Ionic Framework Version: 2.0.0-beta.4
Ionic CLI Version: 2.0.0-beta.24
Ionic App Lib Version: 2.0.0-beta.14
ios-deploy version: 1.8.5 
ios-sim version: 5.0.8 
OS: Mac OS X El Capitan
Node Version: v5.8.0
Xcode version: Xcode 7.3 Build version 7D175 
```

**Comment by dheeran1** [comment]
@adamdbradley Duplicate of https://github.com/driftyco/ionic/issues/6005

**Comment by jgw96** [comment]
Hello @dheeran1! Thanks for all the info and sorry that you are still having this issue! As you can see we have added this to the beta.6 milestone and hope to have it fixed for good in that release.

**Comment by adamdbradley** [comment]
I've updated the docs to describe how to handle multiple nav transitions that have been started asynchronously. The docs are in the process of publishing right now, but this is the gist of it:

### Dismissing And Async Navigation

After an alert has been dismissed, the app may need to also transition
to another page depending on the handler's logic. However, because multiple
transitions were fired at roughly the same time, it's difficult for the
nav controller to cleanly animate multiple transitions that may
have been kicked off asynchronously. This is further described in the
Nav Transition Promises section in the Nav Controller docs. For alerts,
this means it's best to wait for the alert to finish its transition
out before starting a new transition on the same nav controller.

In the example below, after the alert button has been clicked, its handler
waits on async operation to complete, *then* it uses `pop` to navigate
back a page in the same stack. The potential problem is that the async operation
may have been completed before the alert has even finished its transition
out. In this case, it's best to ensure the alert has finished its transition
out first, *then* start the next transition.

```
let alert = Alert.create({
  title: 'Hello',
  buttons: [{
    text: 'Ok',
    handler: () => {
      // user has clicked the alert button
      // begin the alert's dimiss transition
      let navTransition = alert.dismiss();

      // start some async method
      someAsyncOperation(() => {
        // once the async operation has completed
        // then run the next nav transition after the
        // first transition has finished animating out

        navTransition.then(() => {
          this.nav.pop();
        });
      });
      return false;
    }
  }]
});
 *
this.nav.present(alert);
```

It's important to note that the the handler returns `false`. A feature of
button handlers is that they automatically dismiss the alert when their button
was clicked, however, we'll need more control regarding the transition. Because
the handler returns `false`, then the alert does not automatically dismiss
itself. Instead, you now have complete control of when the alert has finished
transitioning, and the ability to wait for the alert to finish transitioning
out before starting a new transition.

docs(alert): dismissing and async navigation:
https://github.com/driftyco/ionic/commit/f1b6d93dce7269d347ce25ffd589b2a73aa2df94

docs(nav): add Nav Transition Promises section: 
https://github.com/driftyco/ionic/commit/5aa816946a0cc5782aff54cc2c9d70ade2c5f0d8
